projects/storefrontlib/src/cms-components/myaccount/payment-methods/payment-methods.component.ts

```diff
@@ -1,124 +1,23 @@
-        // Set first payment method to DEFAULT if none is set
-        if (
-          paymentDetails.length > 0 &&
-          !paymentDetails.find((paymentDetail) => paymentDetail.defaultPayment)
-        ) {
-          this.setDefaultPaymentMethod(paymentDetails[0]);
-        }
-      })
-    );
-
-    this.editCard = null;
-    this.loading$ = this.userPaymentService.getPaymentMethodsLoading();
-    this.userPaymentService.loadPaymentMethods();
-  }
-
-  getCardContent({
-    defaultPayment,
-    accountHolderName,
-    expiryMonth,
-    expiryYear,
-    cardNumber,
-    cardType,
-  }: PaymentDetails): Observable<Card> {
-    return combineLatest([
-      this.translation.translate('paymentCard.setAsDefault'),
-      this.translation.translate('common.delete'),
-      this.translation.translate('paymentCard.deleteConfirmation'),
-      this.translation.translate('paymentCard.expires', {
-        month: expiryMonth,
-        year: expiryYear,
-      }),
-      this.translation.translate('paymentCard.defaultPaymentMethod'),
-    ]).pipe(
-      map(
-        ([
-          textSetAsDefault,
-          textDelete,
-          textDeleteConfirmation,
-          textExpires,
-          textDefaultPaymentMethod,
-        ]) => {
-          const actions: { name: string; event: string }[] = [];
-          if (!defaultPayment) {
-            actions.push({ name: textSetAsDefault, event: 'default' });
-          }
-          actions.push({ name: textDelete, event: 'edit' });
-          const card: Card = {
-            header: defaultPayment ? textDefaultPaymentMethod : null,
-            textBold: accountHolderName,
-            text: [cardNumber, textExpires],
-            actions,
-            deleteMsg: textDeleteConfirmation,
-            img: this.getCardIcon(cardType.code),
-          };
-
-          return card;
-        }
-      )
-    );
-  }
-
-  deletePaymentMethod(paymentMethod: PaymentDetails): void {
-    this.userPaymentService.deletePaymentMethod(paymentMethod.id);
-    this.editCard = null;
-  }
-
-  setEdit(paymentMethod: PaymentDetails): void {
-    this.editCard = paymentMethod.id;
-  }
-
-  cancelCard(): void {
-    this.editCard = null;
-  }
-
-  setDefaultPaymentMethod(paymentMethod: PaymentDetails): void {
-    this.userPaymentService.setPaymentMethodAsDefault(paymentMethod.id);
-  }
-
-  getCardIcon(code: string): string {
-    let ccIcon: string;
-    if (code === 'visa') {
-      ccIcon = this.iconTypes.VISA;
-    } else if (code === 'master' || code === 'mastercard_eurocard') {
-      ccIcon = this.iconTypes.MASTER_CARD;
-    } else if (code === 'diners') {
-      ccIcon = this.iconTypes.DINERS_CLUB;
-    } else if (code === 'amex') {
-      ccIcon = this.iconTypes.AMEX;
-    } else {
-      ccIcon = this.iconTypes.CREDIT_CARD;
-    }
-
-    return ccIcon;
-  }
+export class PaymentMethodsComponent {
+  paymentMethods$ = this.service.get();
+  loading$ = this.service.loading$;
+
+  getCardContent = (paymentMethod) =>
```

Comment:
I've probably set the example myself during an example/call before, but looking at those members I feel they're less readable and less straightforward to extend. Also, some type safety lacking. Can we move to ordinary: 

```ts
getCardContent(paymentMethod: PaymentDetails): Observable<Card> {
   return this.service.getCardContent(paymentMethod);
}
```

If you agree, let's do this in the other PRs as well.

Reply:
I agree and will update the other PR as well